Add user location check for ImageMagick 7

stephank opened the pull request:
On NixOS, like other distributions, headers are a separate package, but unique to NixOS is that they won't be in a shared prefix:

```
/nix/store/9s1wq1033kkg0pqrgfq4hyq9sa4n24hb-imagemagick-7.0.10-27
/nix/store/ypz071bmhbjksphfcb72dh712yp5y6vv-imagemagick-7.0.10-27-dev
```

For NixOS, configure is provided the second (`*-dev`) path (var `IM_IMAGEMAGICK_PREFIX`), so it can pick up `MagickWand-config`. In turn, `MagickWand-config --prefix` will return the first path (var `IM_PREFIX`), and `MagickWand-config --cflags` will list an include path from the second. But header detection in `imagemagick.m4` is generally based on `IM_PREFIX`, which won't work on this case.

It appears there was already a 'user location' case in place for ImageMagick 6, based on `IM_IMAGEMAGICK_PREFIX`. I've duplicated this for ImageMagick 7.

Ideally, we'd need a check based on `--cflags`, I think, but I'm not that good at autoconf.

Danack commented:
> I think, but I'm not that good at autoconf.

Yeah, I know that feeling.

<img width="904" alt="Screen Shot 2020-12-14 at 14 44 19" src="https://user-images.githubusercontent.com/1505719/102094846-b31c9a00-3e1a-11eb-8e6d-7fb693728ba4.png">

> Ideally, we'd need a check based on --cflags,

Why would that be theoretically better? Most people want it to 'just work'™, without needing to twiddle cflags.

Annoyingly, it seems the comment left in that block of m4 config is not correct, so I'll need to actually think about it.

Please could you add a docker file similar to [this one]( 
https://github.com/Imagick/imagick/tree/master/docker/testing7_2) and an entry in the docker-compose.yml so I can test this, rather than committing it blind?

stephank commented:
> Yeah, I know that feeling.
> 
> <img alt="Screen Shot 2020-12-14 at 14 44 19" width="904" src="https://user-images.githubusercontent.com/1505719/102094846-b31c9a00-3e1a-11eb-8e6d-7fb693728ba4.png">

😅

> Why would that be theoretically better? Most people want it to 'just work'™, without needing to twiddle cflags.

Oh, I didn't mean manually setting `CFLAGS`, but detect it based on `MagickWand-config --cflags`, which I assume will always point the right way.

> Annoyingly, it seems the comment left in that block of m4 config is not correct, so I'll need to actually think about it.

Yeah, the confusing part is that its sorting 'user location' before version, I think.

> Please could you add a docker file similar to [this one](https://github.com/Imagick/imagick/tree/master/docker/testing7_2) and an entry in the docker-compose.yml so I can test this, rather than committing it blind?

I've added a commit for it. Hopefully that makes sense? I removed the entry for `ubuntu`, because the Dockerfiles no longer exist for it, and all `docker-compose` invocations fail because of it.

Danack commented:
Thanks. Merged via git.
